lib/posix/pthread_rwlock.c

```diff
@@ -215,8 +215,19 @@ int pthread_rwlock_unlock(pthread_rwlock_t *rwlock)
 static u32_t read_lock_acquire(pthread_rwlock_t *rwlock, s32_t timeout)
 {
 	u32_t ret = 0U;
+	k_timeout_t k_timeout;
+
+	/* Convert integer timeout to k_timeout_t. Code structured in a way
+	 * that a smart compiler may optimize it out (for a common definition
+	 * of K_FOREVER.
+	 */
+	if (timeout == -1) {
```

Comment:
use `SYS_TIMEOUT_MS` instead of rewriting the logic

Reply:
> use SYS_TIMEOUT_MS instead of rewriting the logic

Ack, for this file it indeed makes sense.